src/background/lib/api.js

```diff
@@ -8,7 +8,38 @@ import {
 } from './editorManager'
 import _ from 'lodash'
 
+export const generateDataUri = async (url, callback) => {
```

Comment:
Also, can you make this use promises to stay consistent with the rest of the project?